### ISSUE TITLE
[WR] Use of schema.org

### ISSUE THREAD

**Issue by gkellogg** [w3c/wpub#454]
* [ ] The spec describes the use of `https://schema.org` as a required element of `@context`. This is common practice, but @danbri has expressed frustration for schema.org being part of the execution path of JSON-LD. We recommend that processors cache popular contexts such as schema.org, and you might as well. As the JSON-LD WG is including embedded HTML support, including for contexts, it's possible that in the future, schema.org will not perform content negotiation at `https://schema.org` for `application/ld+json` and will instead include an embedded JSON-LD context in a script element which does something like load `http://schema.org/docs/jsonldcontext.json` through something like the following:
  ```html
  <html><head><script type="application/ld+json">
    {
      "@context": "http://schema.org/docs/jsonldcontext.json"
    }
  </script></head></html>
  ```
  I'm not sure what action you'd take upon something so prospective, particularly given that many specs reference `https://schema.org" (or equivalent), but you should be aware.

* [ ] Some of the examples have invalid JSON syntax, for example the second part of [Example 5](https://w3c.github.io/wpub/#example-5-using-a-text-string-instead-of-a-person-object) is missing a comma (",") after `"type" : "Person"`. The JSON-LD specs have some infrastructure to extract all examples and perform validation of them (both syntactic and semantic), which can find such issues. Of course, this is not simple, but people copy and paste such examples, so it's good that they be validated.

* [ ] In [Example 6](https://w3c.github.io/wpub/#example-6-resource-list-that-includes-one-link-using-a-relative-url-as-a-string-datatypes-svg-and-two-that-display-the-various-properties-of-the-a-linkedresource-object), the "resources" property describes that "datatypes.svg" is treated as a relative URL, however the term definition for "resources" seems inconsistent with this, as when this example is expanded ([playground link](https://json-ld.org/playground/#startTab=tab-expanded&json-ld=%7B%22%40context%22%3A%5B%22https%3A%2F%2Fschema.org%22%2C%22https%3A%2F%2Fwww.w3.org%2Fns%2Fwp-context%22%5D%2C%22type%22%3A%22CreativeWork%22%2C%22resources%22%3A%5B%22datatypes.svg%22%2C%7B%22type%22%3A%22LinkedResource%22%2C%22url%22%3A%22test-utf8.csv%22%2C%22encodingFormat%22%3A%22text%2Fcsv%22%2C%22name%22%3A%22Test%20Results%22%2C%22description%22%3A%22CSV%20file%20containing%20the%20full%20data%20set%20used.%22%7D%2C%7B%22type%22%3A%22LinkedResource%22%2C%22url%22%3A%22terminology.html%22%2C%22encodingFormat%22%3A%22text%2Fhtml%22%2C%22rel%22%3A%22glossary%22%7D%5D%7D&context=%7B%22%40context%22%3A%7B%22vocab%22%3A%22http%3A%2F%2Fexample.com%2Fvocab%2F%22%2C%22property%22%3A%22http%3A%2F%2Fexample.com%2Fvocab%2Fproperty%22%7D%7D)) it is interpreted as a value.

* [ ] [Example 8](https://w3c.github.io/wpub/#example-8-example-accessiblity-metadata-for-a-document-with-text-and-images-the-publication-provides-alternative-text-and-long-descriptions-appropriate-for-each-image-so-can-also-be-read-in-purely-textual-form) uses `ItemList` and `itemListElement`. Note that the values of `itemListElement` are not actually ordered, and order is specified using `itemListOrder` (which I suppose defaults to _Unordered_ or `schema:position` in a `schema:ListItem` value. By using `ItemList` it may give the impression that values of `itemListElement` are ordered, which they are not. You might want to be clear about this.

* [ ] Note that the specification of text direction in JSON-LD is under active discussion (as @iherman well knows) and there is hope that some solution may be forthcoming.

* [ ] The "links" property is described as having string values interpreted as URLs, but when trying this in the playground, they expand to values, not ids. The term definition for "links" should include `"@type": "@id"`.

* [ ] Not schema.org related, but many properties (e.g., "encodingFormat") do not have a language. If it's possible that a publisher might put, say, `"@language": "fr"` in the context, this could cause properties that specifically should _not_ have a language to gain one. You might consider adding `"@language": null` to the term definitions of such properties.

**Comment by iherman** [comment]
Handled the JSON syntax issues in https://github.com/w3c/wpub/commit/4111915349192cbc4122802993a577e8274f8143

**Comment by iherman** [comment]
Actually, the issue with `resources` and `links` are the same (and the same holds for `readingOrder`): to set the `@type` to `@id`, as proposed by @gkellogg. Set in https://github.com/w3c/wp-vocab/commit/be6cadd82414ec36d6968790a8617e0959b4cad8

**Comment by iherman** [comment]
@mattgarrish @avneeshsingh 

> [Example 8](https://w3c.github.io/wpub/#example-8-example-accessiblity-metadata-for-a-document-with-text-and-images-the-publication-provides-alternative-text-and-long-descriptions-appropriate-for-each-image-so-can-also-be-read-in-purely-textual-form) uses ItemList and itemListElement. Note that the values of itemListElement are not actually ordered, and order is specified using itemListOrder (which I suppose defaults to Unordered or schema:position in a schema:ListItem value. By using ItemList it may give the impression that values of itemListElement are ordered, which they are not. You might want to be clear about this.

Can you comment on this? The example comes from the A11y documentation, and I would like to be sure that you are indeed aware of the un-orderness. 

(I agree with @gkellogg that a warning text should be added to the document to make that clear.)

**Comment by iherman** [comment]
> Note that the specification of text direction in JSON-LD is under active discussion (as @iherman well knows) and there is hope that some solution may be forthcoming.

Yep, I am very well aware :-) The whole section may have to be re-worked as a result of that discussion...

(For onlookers, see https://github.com/w3c/rdf-dir-literal/ and the repository issues...)

**Comment by iherman** [comment]
> Not schema.org related, but many properties (e.g., "encodingFormat") do not have a language. If it's possible that a publisher might put, say, "@language": "fr" in the context, this could cause properties that specifically should not have a language to gain one. You might consider adding "@language": null to the term definitions of such properties.

Well, actually, it is schema.org related, because, e.g., `encodingFormat` is a schema.org term (http://schema.org/encodingFormat). But we do have our own additional context, and resetting the language might indeed be a good idea. Will do...

**Comment by mattgarrish** [comment]
> The example comes from the A11y documentation, and I would like to be sure that you are indeed aware of the un-orderness.

Order isn't important for the accessibility properties. It doesn't affect the meaning in any way if you reorder the access modes, sufficient access modes, etc. All that matters it that each set of values is distinct.

**Comment by iherman** [comment]
@mattgarrish will you add some warning note into the text making it clear that order is not important (in contrast to what the example might suggest)?

**Comment by mattgarrish** [comment]
> @mattgarrish will you add some warning note into the text making it clear that order is not important (in contrast to what the example might suggest)?

This is probably better done in the webschemas wiki for the properties, no?

**Comment by iherman** [comment]
@mattgarrish indeed, that may be better.

**Comment by iherman** [comment]
Added `"@language":null` to a number of entries (see https://github.com/w3c/wp-vocab/commit/e12b1eb5f81f5ac779eb43d10aa3d4d94202c66b) in the wpub context, namely:

- accessMode
- accessModeSufficient
- accessibilityFeature
- accessibilityHazard
- duration
- integrity
- encodingFormat
- rel

@gkellogg I wonder whether this should not be suggested to schema.org; most of those properties (except rel and integrity) are schema.org properties...
